pkg/build/controller/build/build_controller.go

```diff
@@ -425,6 +597,22 @@ func (bc *BuildController) createBuildPod(build *buildapi.Build) (*buildUpdate,
 		return update, err
 	}
 
+	// image reference resolution requires a copy of the build
+	build, err = buildutil.BuildDeepCopy(build)
+	if err != nil {
+		return nil, fmt.Errorf("unable to copy build %s: %v", buildDesc(build), err)
+	}
+
+	// Set the output Docker image reference.
+	if err := bc.resolveImageReferences(build); err != nil {
+		// If we cannot resolve the output reference, the output image stream
+		// may not yet exist. The build should remain in the new state and show the
+		// reason that it is still in the new state.
+		update.setReason(buildapi.StatusReasonInvalidImageReference)
+		update.setMessage(buildapi.StatusMessageInvalidImageRef)
+		return update, err
```

Comment:
@csrwng @smarterclayton my understanding is we want to requeue a failure to resolve the output right away, because we have reason to expect it to show up momentarily and we want to reprocess the build until it does.

failure to resolve an input imagestream is, I guess, the same scenario, so it makes sense to me to requeue it immediately.